server/app/services/program/ProgramServiceImpl.java

```diff
@@ -788,6 +789,18 @@ public ProgramDefinition setProgramQuestionDefinitionAddressCorrectionEnabled(
           "Unexpected error: updating a non address question with address correction enabled");
     }
 
+    List<ProgramQuestionDefinition> pqdWithAddressCorrectionEnabled =
```

Comment:
There's perhaps a more efficient way to do this?

Reply:
Suggestion: 
```
if (blockDefinition.programQuestionDefinitions().stream()
        .filter(
            question ->
                question.addressCorrectionEnabled() && question.id() != questionDefinitionId)
        .findAny()
        .isPresent()) {
      throw new ProgramBlockDefinitionContainsInvalidQuestionException(
          programId, blockDefinitionId, questionDefinitionId);
    }
```

Or use what you have in ProgramBlockEditView. Recommend creating a method in BlockDefinition, similar to `hasAddress` to make the check reusable.

Reply:
Actually, I don't think I actually need to collect it to a list anymore.  I think i can use the same logic as in the other place.  This was an experiment.

Reply:
Sorry, didn't reload the page and see your comment.  I'll take a look.

Reply:
Great idea, moved it into BlockDefinition.